great_expectations/execution_engine/partition_and_sample/pandas_data_sampler.py

```diff
@@ -146,7 +146,7 @@ def sample_using_hash(
         hash_function_name: str = self.get_sampling_kwargs_value_or_default(
             batch_spec=batch_spec,
             sampling_kwargs_key="hash_function_name",
-            default_value="md5",
+            default_value="sha256",
```

Comment:
One can already override this in the `batch_spec`. This is not called anywhere internally so I would revert this.